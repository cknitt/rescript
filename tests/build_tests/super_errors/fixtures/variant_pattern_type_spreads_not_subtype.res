type a = One | Two | Three
type b = | ...a | Four | Five
type c = Six | Seven

let lookup = (b: b) =>
  switch b {
  | ...c as c => Console.log(c)
  | Four => Console.log("four")
  | Five => Console.log("five")
  }
