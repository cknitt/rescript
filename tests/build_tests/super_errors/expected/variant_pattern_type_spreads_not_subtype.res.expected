
  [1;31mWe've found a bug for you![0m
  [36m/.../fixtures/variant_pattern_type_spreads_not_subtype.res[0m:[2m7:8[0m

  5 [2mâ”‚[0m let lookup = (b: b) =>
  6 [2mâ”‚[0m   switch b {
  [1;31m7[0m [2mâ”‚[0m   | ...[1;31mc[0m as c => Console.log(c)
  8 [2mâ”‚[0m   | Four => Console.log("four")
  9 [2mâ”‚[0m   | Five => Console.log("five")

  Type c is not a subtype of b